sycl/test-e2e/Matrix/joint_matrix_transposeAB.cpp

```diff
@@ -0,0 +1,133 @@
+//===---joint_matrix_transposeAB.cpp - DPC++ joint_matrix--------------===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+// REQUIRES: aspect-ext_intel_matrix
+
+// RUN: %{build} -o %t.out
+// RUN: %{run} %t.out
+
+// SG size = 32 is not currently supported for SYCL Joint Matrix by IGC on DG2
+// RUN: %if !arch-intel_gpu_dg2 %{ %{build} -o %t_sg32.out -DSG_SZ=32 %}
+// RUN: %if !arch-intel_gpu_dg2 %{ %{run} %t_sg32.out %}
+
+// XFAIL: gpu
+// XFAIL-TRACKER: GSD-5768
+
+// XFAIL: cpu
+// XFAIL-TRACKER: CMPLRLLVM-52693
+
+#include "common.hpp"
+#include <sycl/usm.hpp>
+
+template <typename T, size_t TileRows, size_t TileCols> class MT;
```

Comment:
should "use" be added to the template as well as we may have the three to be the same for both a and b matrices (like 16x16x16 combinations)?

Reply:
I prefer not to add something that is not used currently. It is easy to extend, when it is needed.